server/web/main.go

```diff
@@ -21,29 +21,37 @@ const (
 	restrictedPath = "/restricted"
 )
 
+// Specification represents a map of enviornment variables
+type Specification struct {
+	UserPassword  string `envconfig:"USERPASSWD" required:"true"`
+	EncryptionKey string `split_words:"true" required:"true"`
```

Comment:
Nit: Better to use `envconfig:"ENCRYPTION_KEY"` than split_words (which ends up doing the same thing, but indirectly).

Same bellow.